sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -31,14 +31,20 @@ import org.apache.spark.sql.types.{LongType, StringType, StructType}
 
 class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAndAfter {
```

Comment:
This is an end-to-end test, and we should really reflect the actual behaviors the end users will hit.

The major change I made here is to test with the real v2 session catalog, instead of a fake one.